### PR TITLE
feat: define resources through values

### PR DESCRIPTION
**Pull request by christianknell**
At the moment the manager runs in OOM with the current configuration. With this feature we could override the memory limit through `manager.resources.limits.memory`. The default values are taken from the current Deployment specification.

```yaml
manager:
  ## GKO Manager resource requests and limits
  ## ref: https://kubernetes.io/docs/user-guide/compute-resources/
  ## @param manager.resources.limits The resources limits for the GKO Manager containers
  ## @param manager.resources.requests.memory The requested memory for the GKO Manager containers
  ## @param manager.resources.requests.cpu The requested cpu for the GKO Manager containers
  ##
  resources:
    limits:
      cpu: 500m
      memory: 128Mi
    requests:
      cpu: 5m
      memory: 64Mi
```

### CODE REVIEW

**Comment by fwartner** [comment]
Would love to have that!

**Review by a-cordier — APPROVED** [review]
Thanks for the pull request @christianknell ! Could you rephrase your commit like this ?

`feat: define resources through values`

(We are using semantic release and the release won't get triggered if the commit is not using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)

File: helm/gko/values.yaml [hunk 1]
```diff
@@ -58,6 +58,20 @@ manager:
...
+  ## @param manager.resources.requests.cpu The requested cpu for the GKO Manager containers
+  ##
+  resources:
+    limits:
+      cpu: 500m
+      memory: 128Mi
+    requests:
+      cpu: 5m
+      memory: 64Mi
+
```
Comment (a-cordier) [inline helm/gko/values.yaml:74]: We are using the [bitnami doc generator](https://github.com/bitnami-labs/readme-generator-for-helm) for generating our helm docs, which has a picky linter.

Could you please re-structure the comments like this, run `make helm-reference` and add the result to your commit ?

```yaml
resources:
    limits:
      ## @param manager.resources.limits.cpu The CPU resources limits for the GKO Manager containers
      cpu: 500m
      ## @param manager.resources.limits.memory The memory resources limits for the GKO Manager containers
      memory: 128Mi
    requests:
      ## @param manager.resources.requests.cpu The requested CPU for the GKO Manager containers
      cpu: 5m
      ## @param manager.resources.requests.memory The requested memory for the GKO Manager containers
      memory: 64Mi
```

**Review by a-cordier — CHANGES_REQUESTED** [review]

**Review by christianknell — COMMENTED** [review]
restructured docs according to your comment. But running `make helm-reference` gives me an error:

```bash
npx @bitnami/readme-generator-for-helm -v helm/gko/values.yaml -r helm/gko/README.md
npx: installed 19 in 7.832s
Unexpected token .
make: *** [make/doc.mk:9: helm-reference] Error 1
```

I tried to run `npx @bitnami/readme-generator-for-helm -v helm/gko/values.yaml -r helm/gko/README.md`, but same error:

```bash
npx: installed 19 in 3.115s
Unexpected token .
```

**Comment by a-cordier** [comment]
> restructured docs according to your comment. But running `make helm-reference` gives me an error:
> 
> ```shell
> npx @bitnami/readme-generator-for-helm -v helm/gko/values.yaml -r helm/gko/README.md
> npx: installed 19 in 7.832s
> Unexpected token .
> make: *** [make/doc.mk:9: helm-reference] Error 1
> ```
> 
> I tried to run `npx @bitnami/readme-generator-for-helm -v helm/gko/values.yaml -r helm/gko/README.md`, but same error:
> 
> ```shell
> npx: installed 19 in 3.115s
> Unexpected token .
> ```

@christianknell This might be because of the version of NodeJS you are using. The generator has been tested with [node 16.x and higher](https://github.com/bitnami-labs/readme-generator-for-helm#requirements). Works with 14.x as well from what I've seen

**Comment by fwartner** [comment]
Any updates on this?

**Comment by christianknell** [comment]
@a-cordier @fwartner was able to execute `make helm-reference`. Any more needed?

**Comment by a-cordier** [comment]
> @a-cordier @fwartner was able to execute `make helm-reference`. Any more needed?

Looks all good to me 👌. Could you just rebase all this into one commit with the same headline as put in the PR title ? This way I can make your pull request going through our CI pipeline (right now it is stuck on commit lint). Thanks again for the contribution!

**Comment by a-cordier** [comment]
> > @a-cordier @fwartner was able to execute `make helm-reference`. Any more needed?
> 
> @christianknell Looks all good to me 👌. Could you just rebase all this into one commit with the same headline as put in the PR title ? This way I can make your pull request going through our CI pipeline (right now it is stuck on commit lint). Thanks again for the contribution!

**Comment by christianknell** [comment]
> > > @a-cordier @fwartner was able to execute `make helm-reference`. Any more needed?
> > 
> > 
> > @christianknell Looks all good to me 👌. Could you just rebase all this into one commit with the same headline as put in the PR title ? This way I can make your pull request going through our CI pipeline (right now it is stuck on commit lint). Thanks again for the contribution!

No idea how to do this tbh.

**Comment by a-cordier** [comment]
@christianknell nevermind, I got your branch going through our CI pipeline 👍 i'll just squash your commits on merge and it will be part of the next maintenance release (happening tomorrow)

**Review by a-cordier — APPROVED** [review]

**Comment by christianknell** [comment]
Thanks @a-cordier
